### PR TITLE
*: extract conn and replace `GetAllocator`

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

Just as the title says. It should be reviewed after #6553 is merged.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/utils/grpcutil/grpcutil.go [hunk 1]
```diff
@@ -160,3 +171,57 @@ func GetForwardedHost(ctx context.Context) string {
...
+			Timeout: keepaliveTimeout,
+		}),
+		grpc.WithConnectParams(grpc.ConnectParams{
+			Backoff: backoff.Config{
+				BaseDelay:  time.Second,     // Default was 1s.
+				Multiplier: 1.6,             // Default
+				Jitter:     0.2,             // Default
+				MaxDelay:   3 * time.Second, // Default was 120s.
+			},
+			MinConnectTimeout: 5 * time.Second,
```
Comment (CabinfeverB) [inline pkg/utils/grpcutil/grpcutil.go:196]: Now that we've extracted conn, how about extracting these parameters as well

**Review by CabinfeverB — APPROVED** [review]
LGTM

File: pkg/utils/grpcutil/grpcutil.go (hunk 1 above)
Comment (rleungx) [reply pkg/utils/grpcutil/grpcutil.go:196]: sounds great

**Review by nolouch — APPROVED** [review]

**Pull request merged**